Fix missing global pins and update recipe

ryanvolz opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

This should fix the "not solvable" errors with the [alsa-lib 1.2.6 migration](https://conda-forge.org/status/#alsa_lib126), or at least get us to a place where we can then apply that migration here and fix the downstream not-solvables.

I'm also happy to help maintain this since I'm already involved in related packages (`alsa-lib`, `alsa-plugins`), so I added myself as a maintainer.

ryanvolz committed:
Add unversioned deps to allow global pins to be used.

ryanvolz committed:
Remove run deps that are handled by run_exports.

ryanvolz committed:
Bump build number.

ryanvolz committed:
Add missing openssl dep, used anyway but needed for pin.

ryanvolz committed:
Add libudev dependency, which the build warns as being highly suggested.

Now that we have it on conda-forge, it makes sense to include it.

ryanvolz committed:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.19.0, and conda-forge-pinning 2022.05.23.15.49.11

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ryanvolz committed:
Set udev rule installation dir in conda prefix.

ryanvolz committed:
Remove unused dependencies.

- Need speexdsp not speex for resampling support.
- json-c requirement was dropped in a recent version.
- gettext just doesn't seem to be used.

ryanvolz committed:
Add self to recipe-maintainers to volunteer as a maintainer.

ryanvolz committed:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.19.0, and conda-forge-pinning 2022.05.24.15.15.50

andfoy reviewed (approved):
Thanks for your help with this one!

ryanvolz commented:
@conda-forge-admin, please restart ci

ryanvolz committed:
Move linux-aarch64 and linux-ppc64le to Azure.

TravisCI is misbehaving right now, with the linux-ppc64le build
seemingly never being started.

ryanvolz committed:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.19.0, and conda-forge-pinning 2022.05.25.14.40.58

ryanvolz commented:
@andfoy Sorry I had to add new commits which took away the automerge. Travis wasn't behaving, so I switched to Azure. This is now green and ready to merge.
